### PR TITLE
Add "func" to enaml highlighter

### PR DESCRIPTION
**Pull request by MatthieuDartiailh**
The notion of declarative function have been added in 0.9.6.

### CODE REVIEW

**Comment by SylvainCorlay** [comment]
This seems to supersede your earlier PR.

**Comment by MatthieuDartiailh** [comment]
Yep messed up my branches...

**Comment by SylvainCorlay** [comment]
I guess it is fine to keep only this one open as it is all related to enaml support.

**Comment by MatthieuDartiailh** [comment]
Ok I will close the other.

**Comment by SylvainCorlay** [comment]
Ping @blink1073  re: enaml

File: spyderlib/widgets/projectexplorer.py [hunk 1]
```diff
@@ -1099,7 +1099,7 @@ def show_properties(self, fnames):
             lines += l
         QMessageBox.information(self, _("Project Explorer"),
                                 _("Statistics on source files only:<br>"
-                                  "(Python, C/C++, Fortran)<br><br>"
+                                  "(Python, Cython, IPython, Enaml, C/C++, Fortran)<br><br>"
```
Comment (ccordoba12) [inline spyderlib/widgets/projectexplorer.py]: This line seems to be wider than 79. Please break it.

File: spyderlib/widgets/sourcecode/syntaxhighlighters.py [hunk 2]
```diff
@@ -453,7 +453,8 @@ class CythonSH(PythonSH):
 #==============================================================================
 class EnamlSH(PythonSH):
     """Enaml Syntax Highlighter"""
-    ADDITIONAL_KEYWORDS = ["enamldef", "template", "attr", "event", "const", "alias"]
+    ADDITIONAL_KEYWORDS = ["enamldef", "template", "attr", "event", "const", "alias",
+                           "func"]
```
Comment (ccordoba12) [inline spyderlib/widgets/sourcecode/syntaxhighlighters.py]: What is `func` used for?

Comment (MatthieuDartiailh) [inline spyderlib/widgets/sourcecode/syntaxhighlighters.py]: func is used to declare a func inside an enamldef block.

```python
enamldef MyWidget(PushButton):
    func hello_world():
        print 'Hello world'
    clicked::
         hello_world()
```
Some functions can also be declared as declarative and overridden in the declaration, but I don't remember the syntax. My point here is just that as the language supports it it makes sense for the highlighter to support it too.

Comment (SylvainCorlay) [inline spyderlib/widgets/sourcecode/syntaxhighlighters.py]: `func` is a keyword of enaml?

Comment (MatthieuDartiailh) [inline spyderlib/widgets/sourcecode/syntaxhighlighters.py]: Yes.

**Comment by ccordoba12** [comment]
This change seems very simple, merging :-)

**Pull request merged**